From c95d30ae8e464d44dfd38dfc7689f81562dc08a6 Mon Sep 17 00:00:00 2001
From: Timple Raj M <timple@qti.qualcomm.com>
Date: Wed, 21 May 2025 09:40:19 +0530
Subject: [PATCH] include: soc: qcom: geni-se: Add the QUPV3 Versions

This patch adds the below QUPV3 version macros
QUP_SE_VERSION_1_0
QUP_SE_VERSION_2_5
QUP_SE_VERSION_3_10

Change-Id: I246116a536bd555ea8744c7ddc95e116b81bd83a
Signed-off-by: Timple Raj M <timple@qti.qualcomm.com>
---
 include/soc/qcom/geni-se.h | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/include/soc/qcom/geni-se.h b/include/soc/qcom/geni-se.h
index 698a9256d26..480951f4ed4 100644
--- a/include/soc/qcom/geni-se.h
+++ b/include/soc/qcom/geni-se.h
@@ -259,7 +259,13 @@ enum geni_se_protocol_type {
 #define GENI_SE_VERSION_MINOR(ver) ((ver & HW_VER_MINOR_MASK) >> HW_VER_MINOR_SHFT)
 #define GENI_SE_VERSION_STEP(ver) (ver & HW_VER_STEP_MASK)
 
+/* QUP SE VERSION value for major number 1 and minor number 0 */
+#define QUP_SE_VERSION_1_0		0x10000000
+
 /* QUP SE VERSION value for major number 2 and minor number 5 */
-#define QUP_SE_VERSION_2_5                  0x20050000
+#define QUP_SE_VERSION_2_5		0x20050000
+
+/* QUP SE VERSION value for major number 3 and minor number 10 */
+#define QUP_SE_VERSION_3_10		0x300A0000
 
 #endif
-- 
2.34.1

